### ISSUE TITLE
Do not lint reused variable in trap

### ISSUE THREAD

**Issue by A4-Tacks** [koalaman/shellcheck#3422]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2064
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
trap "rm ${DIR@Q}" int
```

#### Here's what shellcheck currently says:

```
In - line 2:
trap "rm ${DIR@Q}" int
         ^------^ SC2064 (warning): Use single quotes, otherwise this expands now rather than when signalled.
```

#### Here's what I wanted or expected to see:

No warnings
